src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -2832,20 +2832,14 @@ internal bool MethodIsCompatibleWithDelegateOrFunctionPointer(BoundExpression? r
             var methodParameters = method.Parameters;
             int numParams = delegateOrFuncPtrParameters.Length;
 
-            if (methodParameters.Length != numParams + (isExtensionMethod ? 1 : 0))
-            {
-                // This can happen if "method" has optional parameters.
-                Debug.Assert(methodParameters.Length > numParams + (isExtensionMethod ? 1 : 0));
-                Error(diagnostics, getMethodMismatchErrorCode(delegateType.TypeKind), errorLocation, method, delegateType);
-                return false;
-            }
+            Debug.Assert(methodParameters.Length == numParams + (isExtensionMethod ? 1 : 0));
```

Comment:
:memo: `AnalyzeArgumentsForNormalFormNoNamedArguments` uses `CanBeOptional`, which behaves strictly when we're in a conversion. So we can't get here with a mismatch of parameters.